libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/internal/ActiveRouteLineComponentContract.kt

```diff
@@ -0,0 +1,29 @@
+package com.mapbox.navigation.ui.maps.internal
+
+import com.mapbox.navigation.base.ExperimentalPreviewMapboxNavigationAPI
+import com.mapbox.navigation.base.route.NavigationRoute
+import com.mapbox.navigation.core.MapboxNavigation
+import com.mapbox.navigation.core.directions.session.RoutesObserver
+import com.mapbox.navigation.core.lifecycle.MapboxNavigationObserver
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+
+@ExperimentalPreviewMapboxNavigationAPI
+class ActiveRouteLineComponentContract : RouteLineComponentContract, MapboxNavigationObserver {
+    private val _routesFlow = MutableStateFlow<List<NavigationRoute>>(emptyList())
+    private val observer = RoutesObserver { _routesFlow.value = it.navigationRoutes }
+
+    override val navigationRoutes: StateFlow<List<NavigationRoute>> = _routesFlow
+
+    override fun setRoutes(routes: List<NavigationRoute>) {
+        _routesFlow.value = routes
+    }
+
+    override fun onAttached(mapboxNavigation: MapboxNavigation) {
+        mapboxNavigation.registerRoutesObserver(observer)
+    }
+
+    override fun onDetached(mapboxNavigation: MapboxNavigation) {
+        mapboxNavigation.unregisterRoutesObserver(observer)
+    }
+}
```

Comment:
Why are we changing `RouteLineComponentContract` and its default implementation?

This implementation doesn't match previous behavior and will break component installed via "ComponentInstaller.routeLine`.

Reply:
yeah I'm trying some other ways. But the current contract does not allow for observing routes, only setting them.

I was changing the RouteLineComponentContract to make it simpler down stream 🤷 

I can leave it, and handle the oddities in android auto and see what it looks like

Reply:
Yeah, it is a bit more difficult to use `RouteLineComponent` without changing it. But here is the issue I'm not sure how to handle yet

The MapboxNavigation alternative routes are selected by re-ordering the list, in order to select routes without re-ordering the list requires a duplicate route provider for the camera and route line

When selecting the Franklin Street. The camera bounds needs to update to that route, and the list should not be re-ordered.

<img width="808" alt="Screen Shot 2022-06-08 at 3 17 33 PM" src="https://user-images.githubusercontent.com/3021882/172727022-e17f1caa-5a23-456d-8ac1-d6c606cf3048.png">